From 9b9c713200c1d41d56854dd9a80acc9a1614ddf3 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 7 Feb 2024 18:03:36 +0100
Subject: [PATCH 0518/1695] dts: stm32: add the optee-framebuffer memory region
 in stm32mp135f-dk

Add the reserved memory for TUI framebuffer and add
the reference for tzc400 configuration.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a0ba995412e37e03b3b8c5dd7fc86c99836b2e16 dts: stm32: add the optee-framebuffer memory region in stm32mp135f-dk]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I4067810670c2ba53c5518f374fe4aa707750e1a4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/358875
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp135f-dk.dts | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index 684f5b314..0ae2a4431 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -9,8 +9,11 @@
 #include <dt-bindings/clock/stm32mp13-clksrc.h>
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/gpio/stm32mp_gpio.h>
-#include <dt-bindings/soc/stm32mp13-etzpc.h>
 #include <dt-bindings/regulator/st,stm32mp13-regulator.h>
+#include <dt-bindings/soc/stm32mp-tzc400-macro.h>
+#include <dt-bindings/soc/stm32mp13-etzpc.h>
+#include <dt-bindings/soc/stm32mp13-etzpc.h>
+#include <dt-bindings/soc/stm32mp13-tzc400.h>
 #include <dt-bindings/tamper/st,stm32mp13-tamp.h>
 #include "stm32mp135.dtsi"
 #include "stm32mp13xf.dtsi"
@@ -42,7 +45,7 @@
 		optee_framebuffer: optee-framebuffer@dd000000 {
 			/* Secure framebuffer memory */
 			reg = <0xdd000000 0x1000000>;
-			no-map;
+			st,protreg = <TZC_REGION_S_RDWR 0>;
 		};
 	};
 
@@ -573,6 +576,10 @@
 	};
 };
 
+&tzc400 {
+	memory-region = <&optee_framebuffer>;
+};
+
 &uart4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart4_pins_a>;
-- 
2.47.2

